Sources/CoreData/Insert/Client+Insert.swift

```diff
@@ -75,3 +76,51 @@ public extension Client {
 		requestInsert(request, of: type, db: db, queue: queue, jsonTransformer: jsonTransformer, contextObject: contextObject, then: handler)
 	}
 }
+
+@available(iOS 13, *)
+public extension Client {
+	// swiftlint:disable function_default_parameter_at_end
+	/// Shortcut method for building the request, performing an insert, and saving the result.
+	///
+	/// - parameter request:        The router request type
+	/// - parameter type:           The `Insertable` type that will be used in the serialization
+	/// - parameter db:             The database to work in
+	/// - parameter jsonSerializer: The response JSON serializer
+	/// - parameter jsonTransformer: The response JSON transformer
+	/// - parameter contextObject:  The object to pass along to an import operation (see `ImportContext.object`)
+	/// - parameter automaticallyCancelling: Bool determining whether or not the request should be cancelled when the enclosing async context is cancelled.
+	///
+	/// - returns: the result
+	func requestInsert<T: Insertable>(
+		_ request: RouterType,
+		of type: T.Type = T.self,
+		db: DB = DB.shared,
+		jsonOptions: JSONSerialization.ReadingOptions = .allowFragments,
+		jsonTransformer: @escaping (Any) throws -> Any = { $0 },
+		contextObject: Any? = nil,
+		automaticallyCancelling: Bool = false
+	) async -> Result<T, Error> {
+		let response = await self.request(request).responseInsert(
+			of: type,
+			db: db,
+			jsonOptions: jsonOptions,
+			jsonTransformer: jsonTransformer,
+			contextObject: contextObject,
+			automaticallyCancelling: automaticallyCancelling
+		)
+
+		switch response.result {
+		case .success(let value):
+			do {
+				return try await MainActor.run { .success(try value.inContext(db.main)) }
```

Comment:
`main` is deprecated in favour of `view`
```suggestion
				return try await MainActor.run { .success(try value.inContext(db.view)) }
```